_episodes/11-connecting.md

```diff
@@ -0,0 +1,426 @@
+* `-f` (default is /home/user/.ssh/id_algorithm): filename to store your
+  private key. The public key will be identical, with a `.pub` extension added.
+
+```
+{{ site.local.prompt }} ssh-keygen -a 100 -f ~/.ssh/id_ed25519 -t ed25519
+```
+{: .language-bash}
+
+When prompted, enter a strong password that you will remember. There are two
+common approaches to this:
+
+1. Create a memorable passphrase with some punctuation and number-for-letter
+   substitutions, 32 characters or longer. Street addresses work well; just be
+   careful of social engineering or public records attacks.
+2. Use a password manager and its built-in password generator with all
+   character classes, 25 characters or longer. KeePass and BitWarden are two
+   good options.
+
+Note that the terminal will not appear to change while you type the password:
+this is deliberate, for your security. You will be prompted to type it again,
+so don't worry too much about typos.
+
+Take a look in `~/.ssh` (use `ls ~/.ssh`). You should see two new files:
+
+* your private key (`~/.ssh/id_ed25519`): *do not share with anyone!*
+* the shareable public key (`~/.ssh/id_ed25519.pub`): if a system administrator
+  asks for a key, this is the one to send. It is also safe to upload to
+  websites such as GitHub: it is meant to be seen.
+
+> ## No Empty Passwords
+>
+> Nothing is *less* secure than a private key with no password. If you skipped
+> password entry by accident, go back and generate a new key pair *with* a
+> strong password.
+{: .warning}
+
+##### Use RSA for Older Systems
+
+If key generation failed because ed25519 is not available, try using the older
+(but still strong and trustworthy) [RSA][wiki-rsa] cryptosystem. Again, first
+check for an existing key:
+
+```
+{{ site.local.prompt }} ls ~/.ssh/
+```
+{: .language-bash}
+
+If `~/.ssh/id_rsa` already exists, you will need to specify choose a different
+name for the new key-pair. Generate it as above, with the following extra flags:
+
+* `-b` sets the number of bits in the key. The default is 2048.
+  EdDSA uses a fixed key length, so this flag would have no effect.
+* `-o` (no default): use the OpenSSH key format,
+  rather than PEM.
+
+```
+{{ site.local.prompt }} ssh-keygen -a 100 -b 4096 -f ~/.ssh/id_rsa -o -t rsa
+```
+{: .language-bash}
+
+When prompted, enter a strong password that you will remember. There are two
+common approaches to this:
+
+1. Create a memorable passphrase with some punctuation and number-for-letter
+   substitutions, 32 characters or longer. Street addresses work well; just be
+   careful of social engineering or public records attacks.
+2. Use a password manager and its built-in password generator with all
+   character classes, 25 characters or longer. KeePass and BitWarden are two
+   good options.
+
+Take a look in `~/.ssh` (use `ls ~/.ssh`). You should see two new files:
+
+* your private key (`~/.ssh/id_ed25519`): *do not share with anyone!*
```

Comment:
```suggestion
* your private key (`~/.ssh/id_rsa`): *do not share with anyone!*
```

Reply:
ed25519 produces a shorter key with equivalent strength to RSA; in our section immediately above, the ssh-keygen command is `ssh-keygen -a 100 -f ~/.ssh/id_ed25519 -t ed25519`, therefore the keyfile names are correct as-is.